python/cudf/cudf/core/mixins/mixin_factory.py

```diff
@@ -74,8 +75,10 @@ def __get__(self, obj, owner=None):
             return getattr(obj, self._name)
 
 
-def _should_define_operation(cls, operation, base_operation_name):
-    if operation not in dir(cls):
+def _should_define_operation(
+    cls, operation: str, base_operation_name: str, cls_attributes: set[str]
```

Comment:
Can't we remove this now?
```suggestion
    operation: str, base_operation_name: str, cls_attributes: set[str]
```

Reply:
This function still accesses `cls.__dict__` and `cls.__mro__`. We could trade `cls` for 2 additional parameters, but IMO probably not worth it at this point